File: app/src/main/java/com/hfad/news/tsivileva/newschannel/activity/MainActivity.java
package com.hfad.news.tsivileva.newschannel.activity;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;
import androidx.lifecycle.ViewModelStore;
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout;

import android.os.Bundle;

import com.hfad.news.tsivileva.newschannel.R;
import com.hfad.news.tsivileva.newschannel.view.fragments.FragmentFeed;

public class MainActivity extends AppCompatActivity //implements IView {
{

    public static final String logname = "mylog";

    private FragmentTransaction mFragmentTransaction;
    private FragmentFeed fragmentFeed;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.main_activity);

        fragmentFeed=new FragmentFeed();
        mFragmentTransaction= getSupportFragmentManager().beginTransaction();
        mFragmentTransaction.add(R.id.container, fragmentFeed,"feed_fragment");
        mFragmentTransaction.commit();
  }


}

File: app/src/main/java/com/hfad/news/tsivileva/newschannel/activity/MainActivity.kt
package com.hfad.news.tsivileva.newschannel.activity

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.view.menu.MenuBuilder
import androidx.appcompat.widget.Toolbar
import com.hfad.news.tsivileva.newschannel.FEED
import com.hfad.news.tsivileva.newschannel.FEED_CONTENT
import com.hfad.news.tsivileva.newschannel.R
import com.hfad.news.tsivileva.newschannel.view.fragments.FragmentFeeds

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.main_activity)
        supportActionBar?.title=""
        if (supportFragmentManager.findFragmentByTag(FEED) == null) {
            supportFragmentManager.beginTransaction().add(R.id.container, FragmentFeeds(), FEED).commit()
        }

    }

    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<out String>, grantResults: IntArray) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
        val fragment=supportFragmentManager.findFragmentByTag(FEED_CONTENT)

            (fragment as? IPermissionListener)?. getPermissions(requestCode,permissions,grantResults)

    }

}